File: site-keeper-frontend/src/models/Message/MessageRequestModel.ts
import UserRequestModel from "../User/UserRequestModel";

class MessageRequestModel{
    sender: UserRequestModel
    costing: number | null
    message: string
    dateSent: string

    constructor(sender: UserRequestModel,
        costing: number | null,
        message: string,
        dateSent: string){
            this.sender = sender
            this.costing = costing
            this.message = message
            this.dateSent = dateSent
        }
}

export default MessageRequestModel

File: site-keeper-data-access/src/main/java/com/igt/sitekeeper/requestmodel/MessageRequestModel.java
package com.igt.sitekeeper.requestmodel;

import com.igt.sitekeeper.model.Costing;
import com.igt.sitekeeper.model.User;
import lombok.Data;

@Data
public class MessageRequestModel {
    private User sender;
    private Costing costing;
    private String dateSent;
    private String message;
}
